nodeInterface/NodeInterface.go

```diff
@@ -591,7 +591,7 @@ func (n NodeInterface) LegacyLookupMessageBatchProof(c ctx, evm mech, batchNum h
 	return
 }
 
-func (n NodeInterface) blockL1Num(l2BlockNum uint64) (uint64, error) {
+func (n NodeInterface) BlockL1Num(c ctx, evm mech, l2BlockNum uint64) (uint64, error) {
```

Comment:
when we export this function we need to make it available inside nodeinterface contract too since its a precompile. and that should be of the form `"func(*nodeInterface.NodeInterface, *precompiles.Context, *vm.EVM, uint64) (uint64, error)"`

Reply:
Please add documentation explaining that.

Reply:
There's still no documentation added why do we have that extra parameter that's not used anywhere.